### ISSUE TITLE
Getting the model history from the saved model weights.

### ISSUE THREAD

**Issue by moazelshebly** [AntonMu/TrainYourOwnYOLO#159]
First, I would like to thank you for this awesome repo! it really saved me a lot of time.

Now, the repo has worked perfectly for me and I was able to train the model without any problems. However, after finishing the training I realize that the training code saves only the model weights and not all model data. Due to not having a GPU, my model training takes at least a week and I was wondering if there was any way of retrieving the model history from the existing weights file that I have as I need to plot the training and validation losses and the accuracy of the model over time.

I tried a couple of solutions that I found on stackoverflow but still couldn't retrieve the model. I would really appreciate if you have an idea on how to do this or help me in any way possible.

Thanks!

**Comment by AntonMu** [comment]
Hi @moazelshebly - you are welcome. I am glad you got it to work.

One week of training seems very slow - did you try using a free Google Colab notebook with a GPU instead?

For the model files, there are two types of files here. 

One is the final model that was trained on your data which is saved under `TrainYourOwnYOLO/Data/Model_Weights/trained_weights_final.h5`. This is the entire model and is enough to run inference on any other computer to use your model.

The training history is related to the training and validation loss during training and is saved as a tensorflow log each time you train the model. To access the history of the training I recommend using tensorboard by running:

```
tensorboard --logdir Data/Model_Weights
```

Hope this helps!

**Comment by moazelshebly** [comment]
Thanks @AntonMu for the reply.

I am currently using a university GPU to do the training and it is faster but still takes a lot of time because of the large dataset that I have.

I can now see the model history using tensorboard even though I wasn't able to do so before. I had to install version 3.8 of tensorboard to make it work. And regarding the model, I do think that it is enough to run inference using `TrainYourOwnYOLO/Data/Model_Weights/trained_weights_final.h5` but still, it's not the entire model. When I try to load this model in Keras it complains about the missing architecture and when I look at the code in `TrainYourOwnYOLO/2_Training/Train_YOLO.py` I can see that you are only saving the model weights and not the entire model.

Thanks again for the help and for taking the time to implement such a helpful repo!
